### DISCUSSION TITLE
Raw YAML in Pandoc Templates

### DISCUSSION THREAD

**Discussion by jdc-pub** [quarto-dev/quarto-cli#11211] [Q&A]
I'm trying to use [`pubmatter`](https://typst.app/universe/package/pubmatter/) to create a bunch of Typst templates which are compatible with the Quarto author and affiliation schemas. I'd like to expose these Typst templates as custom formats via Quarto extensions.

The easiest way to do so would be to use `pubmatter.load(yaml.decode("$metadata$"))`, where `$metadata$` is a fictional Pandoc template key that returns the raw YAML metadata for a project. 

Does any such key exist?

**Comment by cderv** [comment]
> is a fictional Pandoc template key that returns the raw YAML metadata for a project.

No I don't think so. YAML metadata block is parsed in Lua as `Meta` so you'll find the YAML content as Pandoc metadata but not the raw YAML. For that it would need to be retrieve by reading the YAML header header in a input doc, though YAML metadata blocks can be split. So Looking into `Meta` in AST seems better. 

Why not use `Meta` to get the value you need, and pass it to your function ?
